R/exercise.R

```diff
@@ -1112,6 +1136,12 @@ is_error_result <- function(x) {
   is_exercise_result(x) && length(x$error_message)
 }
 
+# Exercise Prep -----------------------------------------------------------
+
+is_exercise_engine <- function(exercise, engine) {
+  identical(knitr_engine(exercise$engine), tolower(engine))
```

Comment:
```suggestion
  identical(tolower(knitr_engine(exercise$engine)), tolower(engine))
```

Should we `tolower()` both sides of this function so we don't have to care about how the author capitalizes the engine?

Reply:
`knitr_engine()` does that, so we're all good. (`knitr_engine()` is really `tolower(engine %||% "r")`) for consistency though, maybe we should call `kntir_engine()` on both.

```suggestion
  identical(knitr_engine(exercise$engine), knitr_engine(engine))
```

Reply:
Done in c53691ff7ef387646ed13d79d5df7f8cfee3d423